packages/teams-js/src/public/meeting.ts

```diff
@@ -560,4 +585,94 @@ export namespace meeting {
       sendMessageToParent('meeting.appShareButton.setOptions', [shareInformation]);
     }
   }
+
+  /**
+   * Have the external application handle audio (mic & speaker) and turn off host audio.
+   * set isAppHandlingAudio=true in params, will set the Host(Teams) audio to go audioless
+   *   Register the mic mute status change events, and App will be receive the events.
+   * set isAppHandlingAudio=false in params, will restore the Host(Teams) audio
+   *   Deregister the mic mute status change events and App will not be receive the events.
+   *
+   * @throws standard parameter validation error
+   * @param requestAppAudioHandlingParams - RequestAppAudioHandlingParams interface, and contains
+   *   isAppHandlingAudio - true - expect App to handle the audio and Teams to go audioless.
+   *     false - Audio transferred to Teams and Teams to restore the audio.
+   *   callbackMicMuteStateChangedHandler - Callback contains mic status param.
+   * @param callback - Callback contains RequestAppAudioHandlingSdkResponse param, error or operation result.
+   * error can either contain an error of type SdkError (error indication), or null (non-error indication)
+   * result can either contain a true boolean value (successful termination), or null (unsuccessful fetch)
+   * @beta
+   */
+  export function requestAppAudioHandling(
+    requestAppAudioHandlingParams: RequestAppAudioHandlingParams,
+    callback: (error: SdkError | null, isHostAudioless: boolean | null) => void,
```

Comment:
I think I mentioned this last week, but having a callback that returns a nullable error *and* a nullable success value is pretty difficult for consumers to use correctly. Would be better to just always only return the boolean and throw some sort of well formed error (possibly just the one you're getting from the host) rather than returning both. You'll also need some client side error handling here to make sure that we never receive both an error and a boolean at the same time since the host should never send that combination

Reply:
Not sure why you resolved this one: you still have a nullable error and nullable boolean value being returned to developers in the same function

Reply:
@kpyapali *Please* stop resolving this comment without making any code change or any explanation in a response. You still have a nullable sdk error and a nullable boolean being passed to this callback. When you return an SdkError to the caller this way from a function that already throws errors, you've created two different paths that errors can come out of the function. Is there some reason you can't have startAppAudioHandling and stopAppAudioHandling throw the SdkErrors if they exist? That would simplify a lot of this code from the consumer's point of view